File: workflows/pythonSQL2in2out/code/src/main/python/graph/Source0.py
from pyspark.sql import SparkSession
from pyspark.sql.types import *
from pyspark.sql.functions import *
from ..config import Config
from ..prophecylibs import *

@Visual(id = "Source0", label = "Source0", x = 149, y = 50, phase = 0)
@UsesDataset(id = "15", version = 0)
def Source0(spark: SparkSession) -> Source:
    fabric = Config.fabricName
    out = None

    if fabric == "azdb":
        schemaArg = StructType([
            StructField("customer_id", IntegerType(), False),
            StructField("first_name", StringType(), False),
            StructField("last_name", StringType(), False),
            StructField("phone", StringType(), False),
            StructField("email", StringType(), False),
            StructField("country_code", StringType(), False),
            StructField("account_open_date", StringType(), False),
            StructField("account_flags", StringType(), False)
        ])
        out = spark.read\
                  .format("csv")\
                  .schema(schemaArg)\
                  .option("sep", ",")\
                  .option("inferSchema", True)\
                  .load("dbfs:/Users/prophecy/eng/CustomersDatasetInput.csv")
    else:
        raise ValueError("The fabric %s is not supported" % fabric)

    return out

File: workflows/TestEmrWorkflow2/code/src/main/scala/graph/Source0.scala
package graph

import io.prophecy.libs._
import io.prophecy.libs.UDFUtils._
import io.prophecy.libs.Component._
import io.prophecy.libs.DataHelpers._
import io.prophecy.libs.SparkFunctions._
import io.prophecy.libs.FixedFileFormatImplicits._
import org.apache.spark.sql.ProphecyDataFrame._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import config.ConfigStore._
import udfs.UDFs._
import graph._

@Visual(id = "Source0", label = "Source0", x = 242, y = 300, phase = 0)
object Source0 {

  @UsesDataset(id = "201", version = 0)
  def apply(spark: SparkSession): Source = {
    import spark.implicits._

    val fabric = Config.fabricName

    val out = fabric match {
      case "emr" =>
        spark.read
          .format("csv")
          .option("header",      true)
          .option("sep",         ",")
          .option("inferSchema", true)
          .load("s3://abinitio-spark-redshift-testing/Users/prophecy/eng/OrdersDatasetInput.csv")
          .cache()
      case "azdb" =>
        spark.read
          .format("parquet")
          .load("s3://abinitio-spark-redshift-testing/Users/prophecy/eng/OrdersDatasetInput.csv")
          .cache()
      case _ => throw new Exception(s"The fabric '$fabric' is not handled")
    }

    out

  }

}
